db/db_impl/db_impl.cc

```diff
@@ -422,6 +422,11 @@ Status DBImpl::ResumeImpl(DBRecoverContext context) {
     // finish. Those previouly waiting threads can now proceed, which may
     // include closing the db.
     s = error_handler_.ClearBGError();
+    log_write_mutex_.Lock();
+    for (auto& log : logs_) {
+      log.writer.file()->reset_seen_error();
```

Comment:
The code did not compile for me. I think `log.writer` is a pointer.
```suggestion
      log.writer->file()->reset_seen_error();
```

Reply:
I've fixed this and add a test.

Reply:
@jsteemann

Reply:
@liyichao : thanks for your fix! Note that I am not affiliated with the RocksDB team. I was just trying your PR locally.

Reply:
@liyichao I think there were some concerns about the approach on the linked issue: https://github.com/facebook/rocksdb/issues/11643#issuecomment-2184133572